utils/misc.mli

```diff
@@ -38,9 +38,30 @@ val split_last: 'a list -> 'a list * 'a
 val samelist: ('a -> 'a -> bool) -> 'a list -> 'a list -> bool
         (* Like [List.for_all2] but returns [false] if the two
            lists have different length. *)
-
+val sameoption: ('a -> 'a -> bool) -> 'a option -> 'a option -> bool
```

Comment:
this should be `equal_option` -- a more general function would be `compare_option`.